nix/genealogos-module.nix

```
@@ -0,0 +1,30 @@
+{ config, lib, options, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.genealogos;
+in
+{
+  options = {
+    services.genealogos = {
+      enable = mkEnableOption
+        (mdDoc "Genealogos, a Nix sbom generator");
+
+      package = mkPackageOption pkgs "genealogos-api" { };
+    };
+  };
+
+  config = mkIf (cfg.enable) {
+    systemd.services.genealogos =
+      {
+        description = "Genealogos sbom generator";
+        path = [ cfg.package ];
+        wantedBy = [ "multi-user.target" ];
+
+        script = ''
+          ${cfg.package}/bin/genealogos-api
+        '';
```

(nit)

If you don't require this to be a shell script, you can avoid the indirection:

```suggestion
        ServiceConfig.ExecStart = "${cfg.package}/bin/genealogos-api";
```

or slightly funkier:

```suggestion
        ServiceConfig.ExecStart = [ "@genealogos-api" "${cfg.package}/bin/genealogos-api" ];
```

so set `argv[0]` to `genealogos-api` and make it look nicer in htop :D